### ISSUE TITLE
Create a manage.py command to truncate all tables

### ISSUE THREAD

**Issue by cirosantilli** [django-extensions/django-extensions#908]
We currently have `reset_db`, but when I am creating my test databases, I have to run `manage.py migrate` afterwards, and that takes a few seconds.

And even worse, I can't drop tables on Heroku: https://github.com/django-extensions/django-extensions/issues/909

It would be faster if I could just `truncate` all tables instead.

Update: I'm not happy with `python manage.py sqlflush | python manage.py dbshell` because it does a `DELETE`, which could be slower, and does not reset primary key indexes, which makes it harder to setup my test database that uses explicit indexes.

Related: http://stackoverflow.com/questions/2988997/how-do-i-truncate-table-using-django-orm

**Comment by chid** [comment]
Please correct me if I'm wrong. Does `python manage.py sqlflush | python manage.py dbshell` do the job?

**Comment by cirosantilli** [comment]
@chid that does a `DELETE`, which could be slower, and does not reset primary key indexes, which makes it harder to setup my test database that uses explicit indexes.

**Comment by trbs** [comment]
hmm I tested ```sqlflush``` and it does use TRUNCATE and resets sequences for me.
this is using PostGreSQL:

```
BEGIN;
TRUNCATE "auth_permission", "auth_group", "auth_user_user_permissions", "redisboard_redisserver","django_content_type", "django_session", "auth_user_groups",  "django_admin_log", "auth_group_permissions", "auth_user";
SELECT setval(pg_get_serial_sequence('"django_admin_log"','id'), 1, false);
SELECT setval(pg_get_serial_sequence('"auth_permission"','id'), 1, false);
SELECT setval(pg_get_serial_sequence('"auth_group"','id'), 1, false);
SELECT setval(pg_get_serial_sequence('"auth_user"','id'), 1, false);
SELECT setval(pg_get_serial_sequence('"django_content_type"','id'), 1, false);
SELECT setval(pg_get_serial_sequence('"redisboard_redisserver"','id'), 1, false);

COMMIT;
```

**Comment by cirosantilli** [comment]
@trbs ah, I hadn't tried on PostGre, only sqlite locally. On Sqlite Django 1.9.6, it does deletes.

**Comment by Efferon** [comment]
Per SQLite.org -- there is no "truncate" command; most other SQL varieties, including Postgres, MySQL, MSSQL do have "truncate" 

 [SQLite Delete vs Truncate ](https://www.sqlite.org/lang_delete.html) says--- 
The Truncate Optimization

When the WHERE is omitted from a DELETE statement and the table being deleted has no triggers, SQLite uses an optimization to erase the entire table content without having to visit each row of the table individually. This "truncate" optimization makes the delete run much faster. Prior to SQLite version 3.6.5 (2008-11-12), the truncate optimization also meant that the sqlite3_changes() and sqlite3_total_changes() interfaces and the count_changes pragma will not actually return the number of deleted rows. That problem has been fixed as of version 3.6.5 (2008-11-12).

The truncate optimization can be permanently disabled for all queries by recompiling SQLite with the SQLITE_OMIT_TRUNCATE_OPTIMIZATION compile-time switch.

The truncate optimization can also be disabled at runtime using the sqlite3_set_authorizer() interface. If an authorizer callback returns SQLITE_IGNORE for an SQLITE_DELETE action code, then the DELETE operation will proceed but the truncate optimization will be bypassed and rows will be deleted one by one.

**Comment by trbs** [comment]
closing due to inactivity
